modules/service/data-service.js

```diff
@@ -285,6 +285,11 @@ class DataService {
                 if (localQuery) {
                     assertion = await this.createAssertion(nquads);
 
+                    const { assertionId: assertionIdBlockchain } = await this.blockchainService.getAssetProofs(assertion.jsonld.metadata.UALs[0]);
```

Comment:
It looks good to me from feature side. However, let's optimize the code to first check on the blockchain before resolving it